db/blip_sync_context.go

```diff
@@ -576,12 +576,7 @@ func (bsc *BlipSyncContext) sendNoRev(sender *blip.Sender, docID, revID string,
 }
 
 // Pushes a revision body to the client
-func (bsc *BlipSyncContext) sendRevision(sender *blip.Sender, docID, revID string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser) error {
-	var collectionIdx *int
-	if coll, ok := bsc.collections.getIndexForDB(handleChangesResponseCollection); ok {
-		collectionIdx = &coll
-	}
-
+func (bsc *BlipSyncContext) sendRevision(sender *blip.Sender, docID, revID string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser, collectionIdx *int) error {
```

Comment:
I know I wrote this code but I don't understand why we need a new collection here - presumably it's if the user was updated? I wonder if it would be more clear to have a function that calls reloadUser on `DatabaseCollectionWithUser` or just reconstruct it with:

```
collection := &DatabaseCollectionWithUser{DatabaseContext: handleChangesResponseCollection.DatabaseContext, user: bh.user}
```

I think you'd have to lock around user to do this.

Reply:
I have no idea, but would rather not touch it if I don't have to, to be completely honest...

Reply:
Yeah, let's leave this for later.. but I'm pretty sure this is because we have to decide when to reload user information.